wagtail/admin/viewsets/base.py

```diff
@@ -48,6 +48,29 @@ def construct_view(self, view_class, **kwargs):
         filtered_kwargs.update(kwargs)
         return view_class.as_view(**filtered_kwargs)
 
+    def inject_view_methods(self, view_class, method_names):
+        """
+        Check for the presence of any of the named methods on this viewset. If any are found,
+        create a subclass of view_class that overrides those methods to call the implementation
+        on this viewset instead. Otherwise, return view_class unmodified.
+        """
+        viewset = self
+        overrides = {}
+        for method_name in method_names:
+            viewset_method = getattr(viewset, method_name, None)
```

Comment:
Probably not a good idea but we could also handle the issue of "same method name, different signatures depending on the view" by allowing `method_names` to be a dict that maps the view's method name to the viewset's method name (or maybe even the method itself).

Something like

```suggestion
            if isinstance(method_names, dict):
                viewset_method = getattr(viewset, method_names[method_name], None)
                # or if it maps to the method directly, just
                # viewset_method = method_names[method_name]
            else:
                viewset_method = getattr(viewset, method_name, None)
```

So it allows you to have something like `get_queryset_v1` and `get_queryset_v2` on the viewset, and decide which one gets to be the view's `get_queryset` depending on the view.

Reply:
I think it's too early to guess at what kind of helpers will be useful here - that's one reason why this PR is limited to just supporting the one method I need for the API-based-choosers work, rather than trying to pitch this as a general-purpose mechanism for overriding everything.